### DISCUSSION TITLE
Untangle your code with MORE than Service Objects! Meet dry-monads

### DISCUSSION THREAD

**Discussion by swilgosz** [therubyradar/submissions#4] [Submissions]
Anyone knows **how complex Rails controllers can become**. In the new [Hanami Mastery episode](https://hanamimastery.com/episodes/7-untangle-your-app-with-dry-monads), I show how to solve it with Service Objects on Steroids.

**From this episode, You’ll learn:**

- how to write truly framework-agnostic web applications
- read about how I’ve managed to replace Hanami with Rails in less than a day!
- and how you can do the other way around easily.

By an example, you'll be able to see **three gems in action**!
- dry-monads
- dry-matcher
- dry-validation

[Youtube video included!](https://www.youtube.com/watch?v=FnXezvNDnIg)

**Comment by andrewmcodes** [comment]
Included in [Ruby Radar #8](https://rubyradar.dev/issues/ruby-radar-8-rainy-day-edition-695953)
